scripts/histmakers/mz_wlike_with_mu_eta_pt.py

```diff
@@ -217,8 +217,18 @@ def build_graph(df, dataset):
         df = df.Alias("MET_corr_rec_pt", "MET_pt")
         df = df.Alias("MET_corr_rec_phi", "MET_phi")
 
-    df = df.Define("transverseMass", f"wrem::mt_wlike_nano(trigMuons_pt0, trigMuons_phi0, nonTrigMuons_pt0, nonTrigMuons_phi0, MET_corr_rec_pt, MET_corr_rec_phi)")
+    #TODO improve this to include muon mass?
+    ###########
+    # utility plots of transverse mass, with or without recoil corrections
+    ###########
+    met_vars = ("MET_pt", "MET_phi")
+    df = df.Define("transverseMass_uncorr", f"wrem::mt_wlike_nano(trigMuon_pt0, trigMuon_phi0, nonTrigMuon_pt0, nonTrigMuon_phi0, {', '.join(met_vars)})")
```

Comment:
"transverseMass_uncorr" is already defined in L214. Previously I had it only added if recoil is given because otherwise it is the same as "transverseMass". Then of course one has to decide if "uncorr" takes MET or MET_corr_xy but if I understood @jeyserma correctly he wants to use MET_corr_xy for "transverseMass_uncorr"

Reply:
My last commits have modified these lines, the case with no recoil corrections is technically only for tests, so it can be fine to have a duplicate in that case (as I wrote in the new comments there, one can still plot both histograms to verify the are really the same).

I'll polish further after hearing from @jeyserma , I am also removing some of these histograms from the Wmass histmaker since in that case it is hard to debug the goodness of the recoil calibration by plotting MET or mT given the fakes

Reply:
I had second thoughts and let's do the following: the transverseMass_uncorr should point to the MET_pt/phi. For the transverseMass_corr, it should be identical to the uncorrected one when no recoil is applied; it should refer to MET_corr_rec_pt/phi in case the recoil is used.

Reply:
ok, then it is already implemented as @jeyserma said. I just pushed a new commit only to remove some comments